tests/PHPStan/Rules/Arrays/data/internal-classes-overload-offset-access-php84.php

```diff
@@ -0,0 +1,74 @@
+<?php // lint >= 8.4
```

Comment:
This doesn't seem to be working? Not sure why however.

Reply:
This only works in `/nsrt/`. But here you're in a rule test. Which means you need to call `$this->markTestSkipped()` in the test based on PHP_VERSION_ID. There are many existing tests that do this.

Reply:
> This only works in `/nsrt/`. But here you're in a rule test. Which means you need to call `$this->markTestSkipped()` in the test based on PHP_VERSION_ID. There are many existing tests that do this.

Remember I tried to introduce a new annotation syntax that would allow us to use the same skip system for all data files? (main gotcha was data files that required ranges and additional conditions for skipping). The current system is cumbersome and unintuitive. Would you like to revisit my idea? I can provide a sample PR on which to discuss it.

Reply:
The current system works very well in my opinion :)

One day we might be able to test the rules in a similar way we test inferred types (by discovering and filtering files in a directory) but today is not that day.